### PR TITLE
Alterations to app service whitelist scripts +semver: major

### PR DESCRIPTION
**Pull request by cpwevans**
Work for amending the whitelist script for an app service. This now allows the script to be used by a YAML task and to be used within environments that have no access restrictions in place.

### CODE REVIEW

File: Infrastructure-Scripts/Add-AppServiceIpException.ps1 [hunk 1]
```diff
@@ -26,41 +26,44 @@ Param (
...
-
-try {
-    if ($UpdateReleaseName.IsPresent) {
-        $ReleaseName = $env:Release_ReleaseName.Replace("Name", "$env:Release_RequestedFor")
-        Write-Output "##vso[release.updatereleasename]$ReleaseName"
+    if ($RuleName.Length -le 32) {
+        $RuleName = $RuleName.Replace(' ', '')
+    }
+    else {
+        $RuleName = "TEMPORARY_WHITELIST"
```
[the remark this answers is not in the document]
Comment (cpwevans) [reply Infrastructure-Scripts/Add-AppServiceIpException.ps1]: After discussing we have decided to make use of the built-in parameter validation.

File: Infrastructure-Scripts/Add-AppServiceIpException.ps1 [hunk 2]
```diff
@@ -25,42 +25,43 @@ Param (
     [IPAddress]$IpAddress,
     [Parameter(Mandatory = $true)]
     [ValidateNotNull()]
-    [String]$ResourceName,
-    [Parameter(Mandatory = $false)]
-    [Switch]$UpdateReleaseName
+    [String[]]$ResourceName,
```
Comment (nbowes24) [inline Infrastructure-Scripts/Add-AppServiceIpException.ps1:28]: ResourceNames

File: Infrastructure-Scripts/Add-AppServiceIpException.ps1 [hunk 3]
```diff
@@ -11,8 +11,8 @@
     .PARAMETER ResourceName
     The name of the app service
```
Comment (nbowes24) [inline Infrastructure-Scripts/Add-AppServiceIpException.ps1:12]: services?

File: Infrastructure-Scripts/Add-AppServiceIpException.ps1 [hunk 4]
```diff
@@ -11,8 +11,8 @@
...
 
-    .PARAMETER UpdateReleaseName
-    Update the release name on the AzureDevops release pipeline
+    .PARAMETER RuleName
+    The name of the rule being created
 
     .EXAMPLE
     Add-AppServiceIpException -IpAddress 192.168.0.1 -ResourceName das-foobar
```
Comment (nbowes24) [inline Infrastructure-Scripts/Add-AppServiceIpException.ps1:18]: Update example to reflect []

File: Infrastructure-Scripts/Remove-AppServiceIpException.ps1 [hunk 5]
```diff
@@ -22,23 +22,25 @@ Param (
     [IPAddress]$IPAddress,
     [Parameter(Mandatory = $true)]
     [ValidateNotNull()]
-    [String]$ResourceName
+    [String[]]$ResourceName
```
Comment (nbowes24) [inline Infrastructure-Scripts/Remove-AppServiceIpException.ps1:25]: Same comments as the add script

File: Infrastructure-Scripts/Add-AppServiceIpException.ps1 (hunk 4 above)
Comment (cpwevans) [reply Infrastructure-Scripts/Add-AppServiceIpException.ps1]: Updated comments in latest push

File: Infrastructure-Scripts/Add-AppServiceIpException.ps1 (hunk 2 above)
Comment (cpwevans) [reply Infrastructure-Scripts/Add-AppServiceIpException.ps1]: Updated to reflect ability to pass in multiple resource names

File: Infrastructure-Scripts/Remove-AppServiceIpException.ps1 (hunk 5 above)
Comment (cpwevans) [reply Infrastructure-Scripts/Remove-AppServiceIpException.ps1]: Updated to names

**Review by nbowes24 — APPROVED** [review]

**Pull request merged**